File: hoursReport-Client/src/app/Models/hoursReporting.ts
import { Project } from "./project";
import { User } from "./user";

export class HoursReporting {
    hoursReportingId?: number;
    date?: Date;
    begingingTime?: string;
    endTime?: string;
    userId?: number;
    projectId?: number;

    user?: User;
     project?: Project;
}

File: HoursReport-Service/Models/HoursReporting.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace HoursReport_Service
{
    public class HoursReporting
    {
        [Key]
        public int HoursReportingId { get; set; }
        [Required]
        public DateTime Date  { get; set; }
        [Required]
        public string BegingingTime { get; set; }
        [Required]
        public string EndTime { get; set; }
        [Required]
        public int UserId { get; set; }
        [Required]
        public int ProjectId { get; set; }

        public User User { get; set; }
        public Project Project { get; set; }


    }
}
